lib/engine/game/g_1880/game.rb

```diff
@@ -737,17 +737,17 @@ def operating_order
         end
 
         def after_buying_train(train)
-          return if skip_trigger_sr?(train)
+          return unless trigger_sr?(train)
 
           @turn += 1
           @saved_or_round = @round
           @round = new_stock_round
         end
 
-        def skip_trigger_sr?(train)
-          train.name == @depot.upcoming.first.name ||
-          trains_not_triggering_sr?(train.name) ||
-          (train.owner != @depot && @train_marker != train.owner)
+        def trigger_sr?(train)
+          train.name != @depot.upcoming.first.name &&
+          !trains_not_triggering_sr?(train.name) &&
+          (train.owner == @depot || @train_marker == train.owner)
```

Comment:
I'm a little slow today. Please help me understand the logic here.

The train purchased is owned by the depot or the corp with the train marker. Are you trying to determine if it was a cross-buy? If so, I don't follow how the logic is doing that.

Reply:
I think i broke the logic with all the revisions. It doesn't check cross buy as far as i can tell. Let me debug

Reply:
After pr merge of adding source param to buying_train!, i will fix this to:
 train.name != @depot.upcoming.first.name &&
!trains_not_triggering_sr?(train.name) &&
source == @depot